Win8 collation

milgner opened the pull request:
Ideally we'd be able to set the locale during specs, but I haven't found a way to that using WinJS API.
So at the moment, the collation can only be checked in the specs when using german locale setting.

Commit:
Remove obsolete rowcounter function

Commit:
Add Windows-specific locale-aware COLLATE function

pke commented:
from Kraigs book: "If you like, you can allow the user to select the language for the app independent of the system settings. This is done by setting the Windows.Globalization.ApplicationLanguages.primaryLanguageOverride property, as demonstrated in Scenario 10 the  Application resources and localization sample. Scenario 11 also shows loading specific language resources rather than the default."

superquadratic committed:
Use default parameters for length in string conversion functions

superquadratic committed:
Adapt locale collation spec to common SQL style

superquadratic committed:
Use consistent indentation for .then continuations

If they are put in the next line, they now go on the same indentation
level as the async call above.

superquadratic committed:
Make the spec source code pass JSLint

Commit:
Expose collation language to API user

Since changing the display language doesn't seem to change other
settings (usually associated with CultureInfo), I decided to expose
the collation language to the consumer of the API so they can decide
how to sort if they want to use something different from the current
user default.

Specs have been updated to use two different language with
distincitive collation differences.

milgner commented:
This way it seems to work and offers most flexibility

superquadratic committed:
Move member initialization into initialization list

superquadratic committed:
Fix semicolon placement

superquadratic committed:
Adapt locale collation spec to common SQL style

superquadratic commented:
Rebased, (squished a little) and merged in e084198928b70581eaea162a13ad4fc9bc9d3034.
